pandas/tseries/frequencies.py

```diff
@@ -239,17 +239,18 @@ def get_freq(self) -> Optional[str]:
         if not self.is_monotonic or not self.index._is_unique:
             return None
 
-        delta = self.deltas[0]
-        if _is_multiple(delta, _ONE_DAY):
+        delta = min(self.deltas)
```

Comment:
i think the idea behind using deltas[0] was that deltas should be unique at this point.  is that not the case?

Reply:
It is not unique when the index has a business day frequency as you have delays of 1 day or 3 days (for the weekend).
A first version of the bugfix I tried was to check first the unicity and then take deltas[0] to fix the issue with freq=H and DST but it broke the test for business days

Reply:
after reading through the doc of unique_deltas I see the self.deltas area already sorted => no need to take the min.
the key is to check that delta !=0 as with freq="H" and tz with DST, the minimum delta is 0 (in local time)